exporters/zipkin/src/test/java/io/opentelemetry/exporters/zipkin/ZipkinSpanExporterTest.java

```diff
@@ -0,0 +1,224 @@
+import io.opentelemetry.common.AttributeValue;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.data.SpanData.Link;
+import io.opentelemetry.sdk.trace.export.SpanExporter.ResultCode;
+import io.opentelemetry.trace.Span.Kind;
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.Status;
+import io.opentelemetry.trace.TraceFlags;
+import io.opentelemetry.trace.TraceId;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import zipkin2.Call;
+import zipkin2.Endpoint;
+import zipkin2.Span;
+import zipkin2.codec.SpanBytesEncoder;
+import zipkin2.reporter.Sender;
+
+/** Unit tests for {@link ZipkinSpanExporterTest}. */
+@RunWith(MockitoJUnitRunner.class)
+public class ZipkinSpanExporterTest {
+
+  @Mock private Sender mockSender;
+  @Mock private SpanBytesEncoder mockEncoder;
+  @Mock private Call<Void> mockZipkinCall;
+
+  private static final Endpoint localEndpoint =
+      ZipkinSpanExporter.produceLocalEndpoint("tweetiebird");
+  private static final String TRACE_ID = "d239036e7d5cec116b562147388b35bf";
+  private static final String SPAN_ID = "9cc1e3049173be09";
+  private static final String PARENT_SPAN_ID = "8b03ab423da481c5";
+  private static final Map<String, AttributeValue> attributes = Collections.emptyMap();
+  private static final List<SpanData.TimedEvent> annotations =
+      ImmutableList.of(
+          SpanData.TimedEvent.create(
+              1505855799_433901068L, "RECEIVED", Collections.<String, AttributeValue>emptyMap()),
+          SpanData.TimedEvent.create(
+              1505855799_459486280L, "SENT", Collections.<String, AttributeValue>emptyMap()));
+
+  @Test
+  public void generateSpan_remoteParent() {
+    SpanData data = buildStandardSpan().build();
+
+    assertThat(ZipkinSpanExporter.generateSpan(data, localEndpoint))
+        .isEqualTo(buildZipkinSpan(Span.Kind.SERVER, "OK"));
+  }
+
+  @Test
+  public void generateSpan_ServerKind() {
+    SpanData data = buildStandardSpan().setKind(Kind.SERVER).build();
+
+    assertThat(ZipkinSpanExporter.generateSpan(data, localEndpoint))
+        .isEqualTo(buildZipkinSpan(Span.Kind.SERVER, "OK"));
+  }
+
+  @Test
+  public void generateSpan_ClientKind() {
+    SpanData data = buildStandardSpan().setKind(Kind.CLIENT).build();
+
+    assertThat(ZipkinSpanExporter.generateSpan(data, localEndpoint))
+        .isEqualTo(buildZipkinSpan(Span.Kind.CLIENT, "OK"));
+  }
+
+  @Test
+  public void generateSpan_ConsumeKind() {
+    SpanData data = buildStandardSpan().setKind(Kind.CONSUMER).build();
+
+    assertThat(ZipkinSpanExporter.generateSpan(data, localEndpoint))
+        .isEqualTo(buildZipkinSpan(Span.Kind.CONSUMER, "OK"));
+  }
+
+  @Test
+  public void generateSpan_ProducerKind() {
```

Comment:
sorry if I am blind, but definitely add something for the Kind that coerces to null?

Reply:
ah, on the outbound side. Looks like `INTERNAL` ends up with null.  will do!